src/mono/mono/metadata/marshal.c

```diff
@@ -128,6 +130,66 @@ static GENERATE_TRY_GET_CLASS_WITH_CACHE (unmanaged_callconv_attribute, "System.
 
 static gboolean type_is_blittable (MonoType *type);
 
+static IlgenCallbacksToMono ilgenCallbacksToMono = {
+	&mono_get_object_type,
+	&mono_marshal_get_ptr_to_string_conv,
+	&mono_class_is_subclass_of_internal,
+	&mono_class_native_size,
+	&mono_class_try_get_handleref_class,
+	&mono_class_try_get_safehandle_class,
+	&mono_class_try_get_stringbuilder_class,
+	&mono_defaults,
+	&mono_marshal_boolean_conv_in_get_local_type,
+	&mono_marshal_boolean_managed_conv_in_get_conv_arg_class,
+	&mono_marshal_get_ptr_to_stringbuilder_conv,
+	&mono_marshal_get_string_encoding,
+	&mono_marshal_get_string_to_ptr_conv,
+	&mono_marshal_get_stringbuilder_to_ptr_conv,
+	&mono_marshal_load_type_info,
+	&mono_marshal_shared_conv_to_icall,
+	&mono_marshal_shared_emit_marshal_custom_get_instance,
+	&mono_marshal_shared_emit_struct_conv,
+	&mono_marshal_shared_emit_struct_conv_full,
+	&mono_marshal_shared_get_method_nofail,
+	&mono_marshal_shared_get_sh_dangerous_add_ref,
+	&mono_marshal_shared_get_sh_dangerous_release,
+	&mono_marshal_shared_init_safe_handle,
+	&mono_marshal_shared_is_in,
+	&mono_marshal_shared_is_out,
+	&mono_marshal_shared_mb_emit_exception_marshal_directive,
+	&mono_mb_add_local,
+	&mono_mb_emit_add_to_local,
+	&mono_mb_emit_auto_layout_exception,
+	&mono_mb_emit_branch,
+	&mono_mb_emit_branch_label,
+	&mono_mb_emit_byte,
+	&mono_mb_emit_exception,
+	&mono_mb_emit_exception_full,
+	&mono_mb_emit_icall_id,
+	&mono_mb_emit_icon,
+	&mono_mb_emit_ldarg,
+	&mono_mb_emit_ldarg_addr,
+	&mono_mb_emit_ldflda,
+	&mono_mb_emit_ldloc,
+	&mono_mb_emit_ldloc_addr,
+	&mono_mb_emit_managed_call,
+	&mono_mb_emit_op,
+	&mono_mb_emit_stloc,
```

Comment:
This seems like it would break in builds that use the `NOT ENABLE_ILGEN` cmake setting: in that case the methods from `method-builder-ilgen.c` (all the `mono_mb_emit_*` methods) are compiled into a separate `libmono-ilgen.a` static library.  So if you don't like with it (for example, I think wasm and maybe ios, too, don't link it if they don't necessarily link it in) then I think you will get linker errors about undefined symbols here.